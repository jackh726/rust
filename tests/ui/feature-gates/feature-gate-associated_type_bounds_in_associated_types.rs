use std::mem::ManuallyDrop;

trait Tr1 { type As1: Copy; }
trait Tr2 { type As2: Copy; }

struct S1;
#[derive(Copy, Clone)]
struct S2;
impl Tr1 for S1 { type As1 = S2; }

trait _Tr3 {
    type A: Iterator<Item: Copy>;
    //~^ ERROR associated type bounds in associated types are unstable
    //~| ERROR the trait bound `<<Self as _Tr3>::A as Iterator>::Item: Copy` is not satisfied

    type B: Iterator<Item: 'static>;
    //~^ ERROR associated type bounds in associated types are unstable
}

fn main() {}
