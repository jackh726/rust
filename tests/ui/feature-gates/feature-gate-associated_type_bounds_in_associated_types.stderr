error[E0658]: associated type bounds in associated types are unstable
  --> $DIR/feature-gate-associated_type_bounds_in_associated_types.rs:12:22
   |
LL |     type A: Iterator<Item: Copy>;
   |                      ^^^^^^^^^^
   |
   = note: see issue #52662 <https://github.com/rust-lang/rust/issues/52662> for more information
   = help: add `#![feature(associated_type_bounds_in_associated_types)]` to the crate attributes to enable

error[E0658]: associated type bounds in associated types are unstable
  --> $DIR/feature-gate-associated_type_bounds_in_associated_types.rs:16:22
   |
LL |     type B: Iterator<Item: 'static>;
   |                      ^^^^^^^^^^^^^
   |
   = note: see issue #52662 <https://github.com/rust-lang/rust/issues/52662> for more information
   = help: add `#![feature(associated_type_bounds_in_associated_types)]` to the crate attributes to enable

error[E0277]: the trait bound `<<Self as _Tr3>::A as Iterator>::Item: Copy` is not satisfied
  --> $DIR/feature-gate-associated_type_bounds_in_associated_types.rs:12:28
   |
LL |     type A: Iterator<Item: Copy>;
   |                            ^^^^ the trait `Copy` is not implemented for `<<Self as _Tr3>::A as Iterator>::Item`
   |
help: consider further restricting the associated type
   |
LL | trait _Tr3 where <<Self as _Tr3>::A as Iterator>::Item: Copy {
   |            +++++++++++++++++++++++++++++++++++++++++++++++++

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0277, E0658.
For more information about an error, try `rustc --explain E0277`.
